pkg/kubectl/resource/mapper.go

```diff
@@ -54,19 +56,21 @@ func (m *Mapper) InfoForData(data []byte, source string) (*Info, error) {
 	if err != nil {
 		return nil, fmt.Errorf("unable to decode %q: %v", source, err)
 	}
-	var obj runtime.Object
-	var versioned runtime.Object
-	if isThirdParty, gvkOut, err := thirdpartyresourcedata.IsThirdPartyObject(data, gvk); err != nil {
-		return nil, err
-	} else if isThirdParty {
-		obj, err = runtime.Decode(thirdpartyresourcedata.NewDecoder(nil, gvkOut.Kind), data)
-		versioned = obj
-		gvk = gvkOut
-	} else {
-		obj, versioned = versions.Last(), versions.First()
-	}
-	if err != nil {
-		return nil, fmt.Errorf("unable to decode %q: %v [%v]", source, err, gvk)
+
+	obj, versioned := versions.Last(), versions.First()
+	if !m.dynamic {
```

Comment:
So the `dynamic` flag is serving as a behavioral split between using dynamic types or not... for everything. When this is `true`, all objects are being deserialized into `*runtime.Unstructured` or `*runtime.UnstructuredList`. So for create, this whole branch of code doesn't execute. Create behaves the same for thirdparty as it does for pods. This split is here because other commands don't yet use the dynamic codec.

I think I can get rid of the dynamic split in most places by not trying to use the dynamic client. But if the caller want's to use dynamic decoding, I need to somehow prevent the third party from decoding it into it's functionally equivalent object.

Maybe the right answer is a codec that can serialize both unstructured and third party?